proc_fdinfo.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2018 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package procfs
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strings"
+)
+
+// FDInfo contains represents file descriptor information.
+type FDInfo struct {
+	// File descriptor
+	FD string
+	// File offset
+	Pos string
+	// File access mode and status flags
+	Flags string
+	// Mount point ID
+	MntID string
+	// List of inotify lines (structed) in the fdinfo file (kernel 3.8+ only)
+	InotifyInfos []InotifyInfo
+}
+
+// FDInfo constructor. On kernels older than 3.8, InotifyInfos will always be empty.
+func (p Proc) newFDInfo(fd string) (*FDInfo, error) {
```

Comment:
Can you change this to just `FDInfo`?  We did some refactoring recently to remove `new` from the functions where it's not needed.

Reply:
Will try to have this done over the weekend. Do you mean just outright rename this function to `FDInfo`, i.e. 
```
func (p Proc) FDInfo(fd string) ...
```
And should I do the same for `newInotifyInfo` (further down this file)?

Thanks!

Reply:
Right, just rename to `func (p Proc) FDInfo(fd string)`.  You can take a look at some of the other `proc_<thing>.go` files for example.  
I'm thinking `newInotifyInfo` should be `parseInotifyInfo` instead.  It can't be just `InotifyInfo` because then it would conflict with the struct name.